### PR TITLE
Create subctl.tar.gz as well

### PR DESCRIPTION
**Pull request by Jaanki**
along with `tar.xz`'s for all supported platforms and architecture. This will be helpful to

1. add krew install support for subctl
2. add subctl to be downloaded from openshift tools download page directly

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -40,7 +40,7 @@ dockertogoarch = $(patsubst arm/v7,arm,$(1))
 CROSS_TARGETS := linux-amd64 linux-arm64 linux-arm linux-s390x linux-ppc64le windows-amd64.exe darwin-amd64 darwin-arm64
 BINARIES := cmd/bin/subctl
 CROSS_BINARIES := $(foreach cross,$(CROSS_TARGETS),$(patsubst %,cmd/bin/subctl-$(VERSION)-%,$(cross)))
-CROSS_TARBALLS := $(foreach cross,$(CROSS_TARGETS),$(patsubst %,dist/subctl-$(VERSION)-%.tar.xz,$(cross)))
+CROSS_TARBALLS := $(foreach cross,$(CROSS_TARGETS),$(patsubst %,dist/subctl-$(VERSION)-%,$(cross)))
```
Comment (skitt) [inline Makefile]: This switches to phony targets which can result in unnecessary builds; we can easily keep concrete targets so let’s do so.

File: Makefile [hunk 2]
```diff
@@ -89,9 +89,10 @@ bin/lichen:
 cmd/bin/subctl: cmd/bin/subctl-$(VERSION)-$(GOOS)-$(GOARCH)$(GOEXE)
 	ln -sf $(<F) $@
 
-dist/subctl-%.tar.xz: cmd/bin/subctl-%
+dist/subctl-%: cmd/bin/subctl-%
```
Comment (skitt) [inline Makefile]: This whole recipe should be left as-is, and add

```
dist/subctl-%.tar.gz: cmd/bin/subctl-%
	mkdir -p dist
	tar -czf $@ --transform "s/^cmd.bin/subctl-$(VERSION)/" $<
```

File: Makefile [hunk 3]
```diff
@@ -111,6 +112,7 @@ cmd/bin/subctl-%: $(shell find . -name "*.go")
 	components=($$(echo $${target//-/ })); \
 	GOOS=$${components[-2]}; \
 	GOARCH=$${components[-1]}; \
+	#echo $$GOOS $$GOARCH $$components; \
```
Comment (skitt) [inline Makefile]:
```suggestion
```

**Review by skitt — CHANGES_REQUESTED** [review]

File: Makefile (hunk 1 above)
Comment (skitt) [inline Makefile]:
```suggestion
CROSS_TARBALLS := $(foreach cross,$(CROSS_TARGETS),$(patsubst %,dist/subctl-$(VERSION)-%.tar.xz,$(cross))) $(foreach cross,$(CROSS_TARGETS),$(patsubst %,dist/subctl-$(VERSION)-%.tar.gz,$(cross)))
```

**Review by skitt — APPROVED** [review]

**Review by sridhargaddam — APPROVED** [review]

**Review by tpantelis — APPROVED** [review]

**Pull request merged**